Riot/Modules/Application/LegacyAppDelegate.m

```diff
@@ -3726,81 +3726,99 @@ - (void)checkPendingRoomKeyRequestsInSession:(MXSession*)mxSession
         return;
     }
 
+    MXWeakify(self);
     [mxSession.crypto pendingKeyRequests:^(MXUsersDevicesMap<NSArray<MXIncomingRoomKeyRequest *> *> *pendingKeyRequests) {
-
+        
+        MXStrongifyAndReturnIfNil(self);
         NSLog(@"[AppDelegate] checkPendingRoomKeyRequestsInSession: pendingKeyRequests.count: %@. Already displayed: %@",
-              @(pendingKeyRequests.count),
-              roomKeyRequestViewController ? @"YES" : @"NO");
+               @(pendingKeyRequests.count),
+              self->roomKeyRequestViewController ? @"YES" : @"NO");
 
-        if (roomKeyRequestViewController)
+        if (!mxSession.crypto.crossSigning || mxSession.crypto.crossSigning.state != MXCrossSigningStateCrossSigningExists)
```

Comment:
```suggestion
        if (!mxSession.crypto.crossSigning || mxSession.crypto.crossSigning.state == MXCrossSigningStateNotBootstrapped)
```

We do not want to display the modal if the state is `MXCrossSigningStateTrustCrossSigning` or `MXCrossSigningStateCanCrossSign`